### PR TITLE
Collections tests for insights mode

### PR DESCRIPTION
**Pull request by MilanPospisil**
Issue: AAH-2054

https://issues.redhat.com/browse/AAH-2054

### CODE REVIEW

File: .github/workflows/cypress.yml [hunk 1]
```diff
@@ -50,7 +51,18 @@ jobs:
     - name: "Set env.SHORT_BRANCH, COMPOSE_PROFILE"
       run: |
         SHORT_BRANCH=`sed 's/^refs\/heads\///' <<< $BRANCH`
-        COMPOSE_PROFILE=`[ "${{ matrix.test }}" = 'insights' ] && echo 'insights' || echo 'standalone'`
+
+        # COMPOSE_PROFILE=`[ "${{ matrix.test }}" = 'insights' ] && echo 'insights' || echo 'standalone'`
```
Comment (MilanPospisil) [inline .github/workflows/cypress.yml:55]: I dont know how to rewrite it more clearly, any suggestions? expr did not worked with && as expected.

File: .github/workflows/cypress.yml [hunk 2]
```diff
@@ -50,7 +51,18 @@ jobs:
...
+
+        test="${{ matrix.test }}"
+        echo $test
+        if expr "$test" : 'insights*' > /dev/null; then
+          COMPOSE_PROFILE=insights
+          echo "insights"
+        else
+          COMPOSE_PROFILE=standalone
+          echo "standalone"
+        fi
```
[the remark this answers is not in the document]
Comment (himdel) [reply .github/workflows/cypress.yml:65]: Sure :) ..  this should work.. https://github.com/ansible/ansible-hub-ui/pull/3195/files#diff-5da9a41590fb261de0e464c95862528100c42c77dffad53b7c6de228b6afd2bdR54-R65

(without `community` for now)

File: test/cypress/e2e/collections/collection_detail.js [hunk 3]
```diff
@@ -4,13 +4,13 @@ describe('Collection detail', () => {
   const baseURL = `${uiPrefix}repo/published/collection_detail_test_namespace/collection_detail_test_collection`;
 
   function deprecate() {
-    cy.get('[aria-label=Actions]').click();
+    cy.get('main [aria-label=Actions]').click();
```
Comment (himdel) [inline test/cypress/e2e/collections/collection_detail.js:7]: this is most likely wrong for standalone.

Can you use a data-cy instead?

Comment (MilanPospisil) [reply test/cypress/e2e/collections/collection_detail.js:7]: I can, but main does exist in both standalone and insights. Is there any problem with it, does main suppose exist only in insights?

Comment (himdel) [reply test/cypress/e2e/collections/collection_detail.js:7]: For now it does, but any framework screens won't have it .. and then it's not clear what `main` really means and how it's different from not using main. I guess most (or all) mean "pick a first list item with actions", as opposed to actions in a header? Or is this conflicting with something else in insights?

I'm guessing `'[data-cy="CollectionListItem"]:first button[aria-label="Actions"]'`?

Comment (MilanPospisil) [reply test/cypress/e2e/collections/collection_detail.js:7]: Ah you are right, framework screens will not have it. We will have to adjust in future many tests selectors, but that was clear before. No reason to add new future refactor here.

File: .github/workflows/cypress.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (MilanPospisil) [reply .github/workflows/cypress.yml]: ChatGPT told me the same thing :-)

**Review by himdel — APPROVED** [review]
LGTM :+1: 

The screenshot test failure is the same as https://github.com/ansible/ansible-hub-ui/pull/1263#issuecomment-1376573118 ..
so feel free to merge anyway, or it should go away after a rebase :)

**Pull request merged**